app/services/classification/class_param.rb

```diff
@@ -1,41 +0,0 @@
-module Classification
```

Comment:
Let's start from the background:
I want to separate puppet from the core in Foreman. This PR started as a way of getting rid of references to [puppet properties](https://github.com/theforeman/foreman/blob/develop/app/services/classification/base.rb#L3) from `Classification::Base`.
I started reviewing the class, and concluded that inheritance was not the right solution for multiple key types handling here. So I started to get up the stack and look where the `Classification` classes are used, and found that it was the `#info` method. So I refactored it to be extendable and created two extensions that actually used this extension point: one for the static properties provided by the core (`static_info.rb`) and one for puppet properties and parameters (`puppet_info.rb`). The latter also received the functionality yanked form the `Classification` subclasses.

Now about the `Classification` refactoring: `Classification::Base` is a service class that has one main and two secondary methods. The main method is `#values_hash` that receives a set of keys (a scope, to be more accurate) and returns values for each key in the input that are relevant to a single host. Previously the child classes were used to supply the set of keys based on different properties in the host, but now the invocation of the service moved to `puppetclass_parameters` and `smart_variables` methods in `puppet_info.rb`. These methods are responsible to get the relevant keys, pass them to the `Classification::Base` service and format the result. Two secondary services that are served by the `Classification::Base` are `#inherited_values` which returns a subset of `values_hash` and `#value_of_key` which formats the value for a given key. The `#value_of_key` method is a bit interesting, since it uses a cached `SafeRender` instance, that we don't want to create for each value, and that's the main reason that `Classification::Base` class should be instantiated.